### PR TITLE
#1703: The deleted tags are not removed from Tags drop-down until the web page is refreshed

### LINKED ISSUE #1703: The deleted tags are not removed from Tags drop-down until the web page is refreshed
### Preconditions

1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. Uploaded image 

### Steps to reproduce (*)
1. Select the uploaded image by clicking on it
2. Click "three dots", select **Edit**
3. Add three new **Tags**, for example. Click **Save**
4. Click "three dots" again, and select **Edit**
5. Remove the previously added Tags by clicking on the cross (x) near them
6. Click **Save**
7. Click "three dots" again, and select **Edit**

### Expected result (*)
The Tags drop-down is empty
tags3

### Actual result (*)
The Tags drop-down still contains the previously deleted tags
tags4

### Note
The tags are removed from View Details and will disappear after the web page is refreshed. It's just might be confusing for the user. Besides, the previously deleted Tags will be saved to Image Details if click Save after step number 7
**(The Tags are not removed from Database)**

### PR DESCRIPTION
**Pull request by yolouiese**
### Need to verify (*)
- deleted tags have been removed after clicking save

### Description (*)
Replaced the outdated keywords after new image details have been saved

### Fixed Issues (if relevant)
1. Fixes magento/adobe-stock-integration#1703: The deleted tags are not removed from Tags drop-down until the web page is refreshed

### Manual testing scenarios (*)
1. Select the uploaded image by clicking on it
2. Click "three dots", select Edit
3. Add three new Tags, for example. Click Save
4. Click "three dots" again, and select Edit
5. Remove the previously added Tags by clicking on the cross (x) near them
6. Click Save
7. Click "three dots" again, and select Edit

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by yolouiese** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @yolouiese , can you please cover this case with MFTF test

File: app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/image/image-edit.js [hunk 1]
```diff
@@ -223,6 +223,16 @@ define([
...
             return true;
+        },
+
+        /**
+         * Remove cached image details in edit form
+         *
+         * @param {String} id
+         * @param {String} tags
+         */
+        removeCached: function (id, tags) {
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/image/image-edit.js]: I would simply remove cached image and rely on reloading from the backend to make sure frontend and backend are on the same page

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminEnhancedMediaGalleryVerifyUpdatedTagsTest.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminOpenStandaloneMediaGalleryActionGroup" stepKey="openMediaGallery"/>
+        </before>
+        <after>
+            <actionGroup ref="AdminEnhancedMediaGalleryImageDetailsDeleteActionGroup" stepKey="deleteImage"/>
+        </after>
+        <actionGroup ref="AdminEnhancedMediaGalleryUploadImageActionGroup" stepKey="uploadImage">
+            <argument name="image" value="ImageUpload"/>
+        </actionGroup>
+        <actionGroup ref="AdminEnhancedMediaGalleryImageDetailsEditActionGroup" stepKey="editImage"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminEnhancedMediaGalleryVerifyUpdatedTagsTest.xml:31]: I think `AdminEnhancedMediaGalleryViewImageDetails` should be executed before this step

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminEnhancedMediaGalleryVerifyUpdatedTagsTest.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <actionGroup ref="AdminMediaGalleryEditAssetAddKeywordActionGroup" stepKey="setKeywords">
+            <argument name="keyword" value="UpdatedImageDetails.keyword"/>
+        </actionGroup>
+        <actionGroup ref="AdminEnhancedMediaGalleryImageDetailsSaveActionGroup" stepKey="saveImage">
+            <argument name="image" value="UpdatedImageDetails"/>
+        </actionGroup>
+        <actionGroup ref="AdminEnhancedMediaGalleryVerifyImageKeywordsActionGroup" stepKey="verifyAddedKeywords">
+            <argument name="keywords" value="UpdatedImageDetails.keyword"/>
+        </actionGroup>
+        <actionGroup ref="AdminEnhancedMediaGalleryVerifyImageKeywordsActionGroup" stepKey="verifyMetadataKeywords">
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Test/AdminEnhancedMediaGalleryVerifyUpdatedTagsTest.xml]: Can you please also verify the removal of the keyword

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @yolouiese ! Please see my comments

**Comment by yolouiese** [comment]
@magento run all tests

**Comment by yolouiese** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/Test/Mftf/Section/AdminEnhancedMediaGalleryEditDetailsSection.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
         <element name="description" type="textarea" selector="#description"/>
         <element name="newKeyword" type="input" selector="[data-ui-id='keyword']"/>
         <element name="addNewKeyword" type="input" selector="[data-ui-id='add-keyword']"/>
+        <element name="removeSelectedKeyword" type="button" selector="//span[contains(text(), '{{selectedKeywords}}')]/following-sibling::button[@data-action='remove-selected-item']" parameterized="true"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/Section/AdminEnhancedMediaGalleryEditDetailsSection.xml]: I believe only one keyword can be specified here
```suggestion
        <element name="removeSelectedKeyword" type="button" selector="//span[contains(text(), '{{keyword}}')]/following-sibling::button[@data-action='remove-selected-item']" parameterized="true"/>
```

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminMediaGalleryEditAssetRemoveKeywordActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminMediaGalleryEditAssetRemoveKeywordActionGroup">
+        <annotations>
+            <description>Remove Keywords on the Edit Details panel</description>
+        </annotations>
+        <arguments>
+            <argument name="selectedKeywords"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminMediaGalleryEditAssetRemoveKeywordActionGroup.xml]: I'd change the name and specify a type to make it more clear how to use this action group
```suggestion
            <argument name="keyword" type="string"/>
```

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup">
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml]:
```suggestion
    <actionGroup name="AssetAdminEnhancedMediaGalleryAssetDetailsKeywordsAbsentActionGroup">
```

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        <annotations>
+            <description>Verifies removed image keywords on the View Details panel</description>
+        </annotations>
+        <arguments>
+            <argument name="keywords"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml]:
```suggestion
            <argument name="keywords" type="string"/>
```

File: app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup">
+        <annotations>
+            <description>Verifies removed image keywords on the View Details panel</description>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Test/Mftf/ActionGroup/AdminEnhancedMediaGalleryVerifyRemovedImageKeywordsActionGroup.xml]:
```suggestion
            <description>Verifies that the passed comma-separated list of keywords are not present on the View Details panel</description>
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @yolouiese ! Please see my comments on the naming. I think it can be improved a bit to make the action groups easier to reuse

**Comment by yolouiese** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8015](https://jira.corp.magento.com/browse/ENGCOM-8015) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

Tags successfully removed
Peek 2020-08-17 19-21

Manual testing [scenario](https://studio.cucumber.io/projects/131313/test-plan/folders/1337102/scenarios/5064888)

**Pull request merged**